app/javascript/components/ActionItemForm.jsx

```diff
@@ -0,0 +1,403 @@
+            id: this.props.id,
+            title: this.props.title,
+            default: false,
+            completed: false,
+            open: false,
+            editorState: EditorState.createEmpty(),
+            errors: {
+              title: '',
+            },
+            display: this.props.display
+        };
+        this.onChange = editorState => this.setState({ editorState });
+        this.handleClose = this.handleClose.bind(this);
+        this.handleOpen = this.handleOpen.bind(this);
+        this.handleInternalChange = this.handleInternalChange.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    
+    }
+    handleOpen() {
+        this.setState({open: true});
+    }
+    handleClose() {
+        this.setState({ 
+        open: false,
+        internal: this.props.internal,
+        title: this.props.title
+      });
+    //handleInternalChange = 
+    if(this.state.type == 'edit') {
+      this.state.internal = this.props.internal,
+      this.state.title = this.props.title,
+      this.state.description = this.props.description
+    }
+  }
+  checkErrors = field => () => {
+    let errorMessage = '';
+    if (field === 'title') {
+      const { title } = this.state;
+      if (
+        title === '' ||
+        validator.isEmpty(title, { ignore_whitespace: true })
+      ) {
+        errorMessage = 'Title is required';
+      }
+    }
+    this.setState(prevState => ({
+      errors: { ...prevState.errors, [field]: errorMessage },
+    }));
+  };
+
+  handleChange = name => event => {
+    const { value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleInternalChange = name => event => {
+    this.setState(
+      { [name]: event.target.checked }
+    );
+  };
+
+  handleDescriptionChange = name => state => {
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit() {
+    const { type } = this.state;
+
+    let hasErrors = false;
+    Object.keys(this.state.errors).forEach(field => {
+      this.checkErrors(field)();
+      hasErrors = hasErrors || this.state.errors[field] !== '';
+    });
+
+    if (!hasErrors) {
+      if (type === 'create') {
+        const body = {
+          title: this.state.title,
+          description: this.state.description,
+          internal: this.state.internal,
+          participant_id: this.state.participant_id,
+        };
+        
+        apiPost('/api/case_notes', { case_note: body })
+          .then(() => window.location.reload())
+          .catch(error => console.error(error));
+      } else {
+        
+        const newTitle = this.state.title;
+        const newDescription = this.state.tempDescription;
+        const newInternal = this.state.internal;
+
+        this.setState({
+          title: newTitle,
```

Comment:
I also think we don't need to set the state for a `newInternal` and `newTitle` since they probably wouldn't have changed (unless I'm reading the code wrong)